independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProjectDefinition.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.devtools.codestarts;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public interface CodestartProjectDefinition {
```

Comment:
Some javadoc would be nice

Reply:
Yeah I plan to do it all at once: https://github.com/quarkusio/quarkus/issues/12106